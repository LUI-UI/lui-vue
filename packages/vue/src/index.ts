export { default as LuiAvatar } from './components/Avatar/LuiAvatar.vue'
export { default as LuiBadge } from './components/Badge/LuiBadge.vue'
export { default as LuiButton } from './components/Button/LuiButton.vue'
export { default as LuiCheckbox } from './components/Checkbox/LuiCheckbox.vue'
export { default as LuiMenuDropdown } from './components/MenuDropdown/LuiMenuDropdown.vue'
export { default as LuiMenuItem } from './components/MenuItem/LuiMenuItem.vue'
export { default as LuiInput } from './components/Input/LuiInput.vue'
export { default as LuiRadio } from './components/Radio/LuiRadio.vue'
export { default as LuiSelect } from './components/Select/LuiSelect.vue'
export { default as LuiMultiSelect } from './components/MultiSelect/LuiMultiSelect.vue'
export { default as LuiOption } from './components/Option/LuiOption.vue'
export { default as LuiSwitch } from './components/Switch/LuiSwitch.vue'
export { default as LuiTextarea } from './components/Textarea/LuiTextarea.vue'
export { default as LuiModal } from './components/Modal/LuiModal.vue'
export { default as LuiTabGroup } from './components/Tabs/LuiTabGroup.vue'
export { default as LuiTabButtons } from './components/Tabs/LuiTabButtons.vue'
export { default as LuiTabButton } from './components/Tabs/LuiTabButton.vue'
export { default as LuiTabPanels } from './components/Tabs/LuiTabPanels.vue'
export { default as LuiTabPanel } from './components/Tabs/LuiTabPanel.vue'
export { default as LuiTag } from './components/Tag/LuiTag.vue'
export { default as LuiPopover } from './components/Popover/LuiPopover.vue'
export { default as LuiSidebar } from './components/Sidebar/LuiSidebar.vue'
