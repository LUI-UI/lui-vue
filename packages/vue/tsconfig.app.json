{
  "extends": "@vue/tsconfig/tsconfig.web.json",
  "compilerOptions": {
    "composite": true,
    "baseUrl": ".",
    "paths": {
      "@*": ["./src/*"]
    }
  },
  "include": ["env.d.ts", "src/**/*", "src/**/*.vue", "tailwind.config.cjs"],
  "exclude": ["src/**/__tests__/*"]
}
