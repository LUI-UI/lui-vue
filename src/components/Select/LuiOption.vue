<script lang="ts">
import { inject, computed, watch, nextTick, useSlots } from 'vue'
import type { PropType } from 'vue'
import { ContextKey } from './symbols'
import { useId } from '../../utils/useId'
import type { Rounded, Size } from '@/globals/types'
import type { TwClassInterface } from '@/globals/interfaces'

export default {
  name: 'LuiOption',
  inheritAttrs: false
}
</script>
<script setup lang="ts">
const props = defineProps({
  selected: {
    type: Boolean,
    default: false
  },
  disabled: {
    type: Boolean,
    default: false
  },
  size: {
    type: String as PropType<Size>,
    default: 'md'
  },
  rounded: {
    type: [Boolean, String] as PropType<Rounded>,
    default: false
  },
  value: {
    type: [String, Number],
    default: ''
  },
  text: {
    type: String,
    default: ''
  }
})

// const { selectedOption, updateSelectedOption, focusButton } =
//   inject(ContextKey);
const context = inject(ContextKey)
const slots = useSlots()
const optionId = `lui-option-${useId()}`

nextTick(() => {
  // if v-model does not used so we set the initial selectedValue
  if (context?.selectedOption.value === undefined && props.selected && !props.disabled) {
    context?.updateSelectedOption({
      value: props.value,
      text: props.text,
      selected: props.selected
    })
  }
})

watch(
  () => props.selected,
  (value) => {
    context?.updateSelectedOption({
      value: props.value,
      text: props.text,
      selected: value
    })
  }
)

function handleOptionClick(e: any) {
  e.preventDefault()
  context?.updateSelectedOption({
    value: props.value,
    text: props.text,
    selected: props.selected
  })
  nextTick(() => context?.focusButton())
}

const isSelected = computed(() => {
  if (context?.selectedOption.value === undefined) return props.selected
  return typeof context?.selectedOption.value === 'string'
    ? props.text === context?.selectedOption.value
    : props.text === context?.selectedOption.value.text
})

const optionClasses = computed(() => {
  //hover:bg-${props.color}-600/20
  const classes: TwClassInterface = {
    backgroundColor: props.disabled
      ? ''
      : isSelected.value === true
      ? 'bg-primary-500 hover:text-primary-600 focus:text-primary-600'
      : 'hover:bg-primary-500/20 focus:bg-primary-500/20',
    padding: {
      'px-1.5 py-1': props.size === 'xs',
      'px-2 py-1.5': props.size === 'sm',
      'px-2 py-2.5': props.size === 'md',
      'p-2.5': props.size === 'lg',
      'px-3 py-3.5': props.size === 'xl'
    },
    textColor: props.disabled
      ? 'text-secondary-300 dark:text-secondary-700'
      : isSelected.value === true
      ? 'text-white'
      : 'text-secondary-600 dark:text-secondary-300 hover:text-primary-500 focus:text-primary-500',
    outlineWidth: 'outline-none',
    // margin: props.size === "xs" || props.size === "sm" ? "mb-1.5" : "mb-2",
    borderRadius: {
      'rounded-md': props.rounded === true,
      'rounded-2xl': props.rounded === 'full'
    },
    display: slots.prepend || slots.append ? 'flex' : '',
    alignItems: slots.prepend || slots.append ? 'items-center' : '',
    cursor: props.disabled ? 'cursor-not-allowed' : '',
    pointerEvents: props.disabled ? 'pointer-events-none' : ''
  }
  return Object.values({ ...classes })
})
</script>
<template>
  <li
    role="option"
    tabindex="-1"
    v-bind="$attrs"
    :id="optionId"
    :aria-selected="isSelected"
    :class="optionClasses"
    @click="handleOptionClick"
  >
    <span v-if="$slots.prepend"> <slot name="prepend" /> </span>
    <span :class="$slots.prepend ? 'ml-1' : ''"> {{ text }} </span>
    <span v-if="$slots.append" class="ml-auto">
      <slot name="append" />
    </span>
    <!-- <span><slot /></span> -->
  </li>
</template>
