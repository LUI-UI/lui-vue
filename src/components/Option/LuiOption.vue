<script lang="ts">
import { computed, inject, nextTick, ref, useSlots, watch } from 'vue'
import type { PropType } from 'vue'
import { ContextKey } from '../Select/symbols'
import { useId } from '../../utils/useId'
import type { Rounded, Size } from '@/globals/types'
import type { TwClassInterface } from '@/globals/interfaces'

export default {
  name: 'LuiOption',
  inheritAttrs: false,
}
</script>

<script setup lang="ts">
const props = defineProps({
  selected: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  size: {
    type: String as PropType<Size>,
    default: 'md',
  },
  rounded: {
    type: [Boolean, String] as PropType<Rounded>,
    default: false,
  },
  value: {
    type: [String, Number],
    default: '',
  },
  text: {
    type: String,
    default: '',
  },
})

// const { selectedOption, updateSelectedOption, focusButton } =
//   inject(ContextKey);
const context = inject(ContextKey)
const slots = useSlots()
const optionId = `lui-option-${useId()}`
const optionRef = ref(null)
const isKeyboardHovered = computed(() => optionId === context?.currentId.value)
nextTick(() => {
  // if v-model does not used so we set the initial selectedValue
  if (context?.selectedOption.value === undefined && props.selected && !props.disabled) {
    context?.updateSelectedOption({
      value: props.value,
      text: props.text,
      selected: props.selected,
    })
  }
})

watch(
  () => props.selected,
  (value) => {
    context?.updateSelectedOption({
      value: props.value,
      text: props.text,
      selected: value,
    })
  },
)

function handleOptionClick(e: any) {
  e.preventDefault()
  const isSelectMultiple = Array.isArray(context?.selectedOption.value)
  if (isSelectMultiple)
    e.stopPropagation()
  context?.updateSelectedOption({
    value: props.value,
    text: props.text,
    selected: props.selected,
  })
  nextTick(() => context?.focusButton())
}
function anySelectedOption() {
  if (Array.isArray(context?.selectedOption.value))
    return context?.selectedOption.value.length !== 0
  return context?.selectedOption.value.text !== ''
}
function isSelectedsIncludesOption() {
  if (!Array.isArray(context?.selectedOption.value))
    return false
  return context?.selectedOption.value.includes(props.text) || context?.selectedOption.value.includes(props.value as string)
}
const isSelected = computed(() => {
  if (!anySelectedOption())
    return props.selected
  return Array.isArray(context?.selectedOption.value)
    ? isSelectedsIncludesOption()
    : context?.selectedOption.value.text === props.text
})
const optionClasses = computed(() => {
  // hover:bg-${props.color}-600/20
  const classes: TwClassInterface = {
    // backgroundColor: props.disabled
    //   ? ''
    //   : isSelected.value
    //     ? isKeyboardHovered.value
    //       ? 'bg-danger-500'
    //       : 'hover:bg-primary-500/20 focus-visible:bg-primary-500/20'
    //     : '',
    // default: '', hover
    // selected: '', hover
    // keybpardHover: selected ? '' : ''

    backgroundColor: props.disabled
      ? ''
      : {
          'hover:bg-primary-500/20': !isSelected.value,
          'bg-primary-500 hover:bg-primary-500/80 focus-visible:bg-primary-500/80': isSelected.value && !isKeyboardHovered.value,
          'bg-primary-500/20': isKeyboardHovered.value && !isSelected.value,
          'bg-primary-500/80': isKeyboardHovered.value && isSelected.value,
        },
    padding: {
      'py-1 px-1.5': props.size === 'xs',
      'py-1.5 px-2': props.size === 'sm',
      'py-2 px-2.5': props.size === 'md',
      'py-2.5 px-2.5': props.size === 'lg',
      'py-3.5 px-3': props.size === 'xl',
    },
    fontSize: {
      'text-xs': props.size === 'xs',
      'text-sm': props.size === 'sm',
      'text-base': props.size === 'md',
      'text-lg': props.size === 'lg',
      'text-xl': props.size === 'xl',
    },
    textColor: props.disabled
      ? 'text-secondary-300 dark:text-secondary-700'
      : {
          'text-secondary-600 dark:text-secondary-300 hover:text-primary-500 focus-visible:text-primary-500': !isKeyboardHovered.value && !isSelected.value, // default state
          'text-white': isSelected.value, // selected
          'text-primary-500': isKeyboardHovered.value && !isSelected.value, // keyboard-hovered && not-selected
        },
    outlineWidth: 'outline-none',
    borderRadius: {
      'rounded-md': props.rounded === true,
      'rounded-2xl': props.rounded === 'full',
    },
    display: slots.prepend || slots.append ? 'flex' : '',
    alignItems: slots.prepend || slots.append ? 'items-center' : '',
    cursor: props.disabled ? 'cursor-not-allowed' : 'cursor-pointer',
    pointerEvents: props.disabled ? 'pointer-events-none' : '',
  }
  return Object.values({ ...classes })
})
defineExpose({
  el: optionRef,
})
</script>

<template>
  <li
    v-bind="$attrs"
    :id="optionId"
    ref="optionRef"
    role="option"
    tabindex="-1"
    :aria-selected="isSelected"
    class="lui-option"
    :class="optionClasses"
    @click="handleOptionClick"
  >
    <span v-if="$slots.prepend"> <slot name="prepend" /> </span>
    <span :class="$slots.prepend ? 'ml-1' : ''"> {{ text }} </span>
    <span v-if="$slots.append" class="ml-auto">
      <slot name="append" />
    </span>
  </li>
</template>
