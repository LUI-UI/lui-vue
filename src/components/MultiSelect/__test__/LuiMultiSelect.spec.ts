import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import type { DOMWrapper, VueWrapper } from '@vue/test-utils'
import { mount } from '@vue/test-utils'
import { ref } from 'vue'
import LuiOption from '../../Option/LuiOption.vue'
import LuiMultiSelect from '../LuiMultiSelect.vue'

const names = [
  {
    value: 1,
    text: 'Wade Cooper',
  },
  {
    value: 2,
    text: 'Arlene Mccoy',
  },
  {
    value: 3,
    text: 'Devon Webb',
  },
  {
    value: 4,
    text: 'Tom Cook',
  },
  {
    value: 5,
    text: 'Tanya Fox',
  },
  {
    value: 6,
    text: 'Hellen Schmidt',
  },
]

describe('Default LuiMultiSelect', () => {
  let wrapper: VueWrapper<any>
  const defaultProps = {
    rounded: true,

    block: false,

    state: null,

    stateIcon: null,

    options: names,

    placeholder: '',

    size: 'md',

    description: null,

    searchable: false,

    tags: true,

    disabled: false,
    tagProps: undefined,

    hideAppend: false,

    menuPosition: 'bottomLeft',

    modelValue: undefined,
  }
  beforeEach(() => {
    wrapper = mount(LuiMultiSelect, {
      props: {
        options: names,
      },
    })
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })
  it('renders with default props properly', async () => {
    await wrapper.vm.$nextTick
    expect(wrapper.props()).toEqual(defaultProps)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html).toMatchSnapshot()
  })
  it('renders LuiTag component as default', async () => {
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')

    const selecteds = wrapper.findAll(
      '[data-testid="lui-multi-select-selected"]',
    )
    expect(selecteds[0].element.tagName).toBe('BUTTON')
    expect(selecteds[0].classes()).contain('lui-tag')
  })
  it('renders LuiText when tags prop is false', async () => {
    wrapper.setProps({ tags: false })
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')

    const selecteds = wrapper.findAll(
      '[data-testid="lui-multi-select-selected"]',
    )
    expect(selecteds[0].element.tagName).toBe('SPAN')
  })

  it('closes options when user click outside', async () => {
    let optionsContainer = wrapper.find(
      '[data-testid="lui-multi-select-options-container"]',
    )
    const mainContainer = wrapper.find(
      '[data-testid="lui-multi-select-container"]',
    )
    expect(optionsContainer.attributes().style).toBe('display: none;')

    await mainContainer.trigger('click')

    optionsContainer = wrapper.find(
      '[data-testid="lui-multi-select-options-container"]',
    )
    expect(optionsContainer.attributes().style).not.toBe('display: none;')
    // simulate the outside click
    await document.body.click()
    optionsContainer = wrapper.find(
      '[data-testid="lui-multi-select-options-container"]',
    )
    expect(optionsContainer.attributes().style).toBe('display: none;')
  })
})

describe('LuiMultiSelect with v-model', () => {
  let wrapper: VueWrapper<any>
  const myRef = ref([names[2], names[3]])

  beforeEach(() => {
    wrapper = mount(LuiMultiSelect, {
      props: {
        'options': names,
        'onUpdate:modelValue': (e: any) => (myRef.value = e),
      },
    })
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })

  it('should add selected options to v-model', async () => {
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')
    expect(myRef.value).toEqual(['Wade Cooper'])
  })

  it('should remove selected options from v-model', async () => {
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')
    expect(myRef.value).toEqual(['Wade Cooper'])

    // Click again to remove the selection
    await options[0].trigger('click')
    expect(myRef.value).toEqual([])
  })

  it('should update selected options in the UI', async () => {
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')
    let selecteds = wrapper.findAll(
      '[data-testid="lui-multi-select-selected"]',
    )
    expect(selecteds.length).toBe(1)
    expect(selecteds[0].text()).toBe(names[0].text)

    // Click again to remove the selection
    await options[0].trigger('click')
    selecteds = wrapper.findAll('[data-testid="lui-multi-select-selected"]')
    expect(selecteds.length).toBe(0)
  })
})
describe('LuiMultiSelect with slots', () => {
  let wrapper: VueWrapper<any>
  const names = [
    { value: 1, text: 'Wade Cooper' },
    { value: 2, text: 'Arlene Mccoy' },
    { value: 3, text: 'Devon Webb' },
    { value: 4, text: 'Tom Cook' },
    { value: 5, text: 'Tanya Fox' },
    { value: 6, text: 'Hellen Schmidt' },
  ]

  const testTemplate = {
    components: { LuiMultiSelect, LuiOption },
    template: `
      <lui-multi-select>
        <lui-option v-for="item in options" :key="item.value" :text="item.text" data-testid="lui-multi-select-option" />
      </lui-multi-select>
    `,
    data() {
      return {
        options: names,
      }
    },
  }

  beforeEach(() => {})

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })

  it('should render options correctly', () => {
    wrapper = mount(testTemplate)
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    expect(options.length).toBe(names.length)
  })

  it('should render prepend slot correctly', () => {
    wrapper = mount(LuiMultiSelect, {
      props: { options: names },
      slots: {
        prepend: `
        <i class="ri-home-line"></i>
     `,
      },
    })
    expect(
      wrapper.find('[data-testid="lui-multi-select-slot-prepend"]').exists(),
    ).toBe(true)
  })
  it('should render append slot correctly', () => {
    wrapper = mount(LuiMultiSelect, {
      props: { options: names },
      slots: {
        append: `
        <i class="ri-home-line"></i>
     `,
      },
    })
    expect(
      wrapper.find('[data-testid="lui-multi-select-slot-append"]').exists(),
    )
  })
})

describe('LuiMultiSelect keyboard events', () => {
  let wrapper: VueWrapper<any>
  let mainContainer: DOMWrapper<Element>
  let simulateKeyPress: (arg0: string) => void

  beforeEach(async () => {
    wrapper = await mount(LuiMultiSelect, {
      props: {
        options: names,
        searchable: true,
      },
    })
    mainContainer = wrapper.find('[data-testid="lui-multi-select-container"]')

    simulateKeyPress = async (key: string) =>
      await wrapper
        .find('[data-testid="lui-multi-select-options-container"]')
        .trigger('keydown', { code: key })
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })

  it('slides on the dropdown menu by arrow keys', () => {
    // Simulate arrow down key press

    simulateKeyPress('ArrowDown')
    expect(wrapper.vm.listboxState.currentIndex).toBe(1)
    simulateKeyPress('ArrowDown')
    expect(wrapper.vm.listboxState.currentIndex).toBe(2)
    simulateKeyPress('ArrowUp')
    expect(wrapper.vm.listboxState.currentIndex).toBe(1)
  })
  it('selects the option by enter key', async () => {
    // check if selecteds count increasing when enter key is pressed
    let selecteds = wrapper.findAll(
      '[data-testid="lui-multi-select-selected"]',
    )
    expect(selecteds.length).toBe(0)
    await simulateKeyPress('Enter')
    console.log(wrapper.vm.selectedOptions)
    selecteds = wrapper.findAll('[data-testid="lui-multi-select-selected"]')
    expect(selecteds.length).toBe(1)
  })
  it('updates the focused index to 0 with home key', () => {
    // give a random value other than 0
    wrapper.vm.listboxState.currentIndex = 2

    simulateKeyPress('Home')
    expect(wrapper.vm.listboxState.currentIndex).toBe(0)
  })
  it('updates the focused index to last index with end key', () => {
    simulateKeyPress('End')
    expect(wrapper.vm.listboxState.currentIndex).toBe(
      wrapper.props().options.length - 1,
    )
  })
  it('closes the listbox with the escape key', () => {
    wrapper.vm.optionsActive = true
    expect(wrapper.vm.optionsActive).toBe(true)
    simulateKeyPress('Escape')
    expect(wrapper.vm.optionsActive).toBe(false)
  })
})
describe('LuiMultiSelect Search Feature', () => {
  let wrapper: VueWrapper<any>
  let inputEl: DOMWrapper<Element>
  let optionsContainer: DOMWrapper<Element>

  beforeEach(async () => {
    wrapper = await mount(LuiMultiSelect, {
      props: {
        options: names,
        searchable: true,
      },
    })
    inputEl = wrapper.find('[data-testid="lui-multi-select-input"]')
    optionsContainer = wrapper.find(
      '[data-testid="lui-multi-select-options-container"]',
    )
  })

  it('should filter options based on the search query, and select from the query', async () => {
    await inputEl.setValue('tanya')
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    expect(options.length).toBe(1)

    // selection from the query

    await optionsContainer.trigger('keydown', { code: 'ArrowDown' })
    await optionsContainer.trigger('keydown', { code: 'Enter' })
    const selecteds = wrapper.findAll(
      '[data-testid="lui-multi-select-selected"]',
    )
    expect(selecteds.length).toBe(1)
  })

  it('should display all options when search query is cleared', async () => {
    await inputEl.setValue('')
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    expect(options.length).toBe(names.length)
  })

  it('should handle search with selected items', async () => {
    await optionsContainer.trigger('keydown', { code: 'ArrowDown' })
    await optionsContainer.trigger('keydown', { code: 'Enter' })
    await inputEl.setValue('devon')
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    expect(options.length).toBe(1)
  })
  it('should keep text field focused while selecting items from open menu', async () => {
    const wrapper = await mount(LuiMultiSelect, {
      props: {
        options: names,
        searchable: true,
      },
      attachTo: document.body,
    })
    expect(optionsContainer.attributes().style).toBe('display: none;')

    const mainContainer = wrapper.find(
      '[data-testid="lui-multi-select-container"]',
    )
    await mainContainer.trigger('click')

    optionsContainer = wrapper.find(
      '[data-testid="lui-multi-select-options-container"]',
    )
    expect(optionsContainer.attributes().style).not.toBe('display: none;')

    // Simulate ArrowDown key press
    await optionsContainer.trigger('keydown', { code: 'ArrowDown' })

    // Simulate Enter key press
    await optionsContainer.trigger('keydown', { code: 'Enter' })

    // Update inputEl
    inputEl = wrapper.find('[data-testid="lui-multi-select-input"]')
    expect(document.activeElement).toBe(inputEl.element)
  })
  afterEach(() => {
    // Clean up, reset, or destroy the wrapper as needed
    wrapper.unmount()
  })
})
// tests for placeholder
describe('Default LuiMultiSelect', () => {
  let wrapper: VueWrapper<any>
  beforeEach(() => {
    wrapper = mount(LuiMultiSelect, {
      props: {
        options: names,
        placeholder: 'names',
      },
    })
  })

  afterEach(() => {
    // Clean up after each test
    wrapper.unmount()
  })
  it('displays the placeholder when there is no selected', () => {
    const container = wrapper.find(
      '[data-testid="lui-multi-select-container"]',
    )
    expect(container.text()).toBe(wrapper.props().placeholder)
  })
  it('does not display the placeholder when there are some selecteds', async () => {
    // click the first option
    const options = wrapper.findAll('[data-testid="lui-multi-select-option"]')
    await options[0].trigger('click')
    const container = wrapper.find(
      '[data-testid="lui-multi-select-container"]',
    )

    expect(container.text()).not.toBe(wrapper.props().placeholder)
  })
  it('renders the placeholder as the first element in dropdown menu', async () => {
    const placeholder = wrapper.find(
      '[data-testid="lui-multi-select-placeholder"]',
    )
    expect(placeholder.exists()).toBe(true)
    expect(placeholder.text()).toBe(wrapper.props().placeholder)
  })
})
