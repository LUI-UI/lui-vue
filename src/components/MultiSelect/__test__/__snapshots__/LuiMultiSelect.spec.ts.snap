// Vitest Snapshot v1

exports[`Default LuiMultiSelect > renders with default props properly 1`] = `[Function]`;
